xwiki-platform-core/xwiki-platform-attachment/xwiki-platform-attachment-validation/xwiki-platform-attachment-validation-api/src/main/java/org/xwiki/attachment/validation/AttachmentValidator.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.attachment.validation;
+
+import java.io.InputStream;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import javax.servlet.http.Part;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Provide the operations to validate an attachment. For instance, by checking the size or the mimetype of the
+ * attachment.
+ *
+ * @version $Id$
+ * @since 14.10RC1
+ */
+@Role
+@Unstable
+public interface AttachmentValidator
+{
+    /**
+     * Check if the part is a valid attachment in the current space.
+     *
+     * @param part the part to validate
+     * @throws AttachmentValidationException in case of error when validating the part
+     */
+    void validateAttachment(Part part) throws AttachmentValidationException;
```

Comment:
I think I'd have also provided an API taking an XWikiAttachment for extensions that directly manipulates the attachments (thinking about CR)

Reply:
I can't because otherwise we have a circular dependency between this module and oldcore. The alternative is to move this interface in oldcore (but that might lead to other issues).

Reply:
It should not be too hard for CR to use `#validateAttachment(long attachmentSize, Supplier<Optional<InputStream>> supplier, String filename)`. Even `#validateAttachment(Part)` is actually just a helper here, and I'm wondering if it's really correct to see something specific to the servlets in the AttachmentValidator API.

Reply:
Done, see https://github.com/xwiki/xwiki-platform/pull/1959/files#diff-b3291e0da5c23912c8ca043e8ce77635486cf9ef433b28d6a7db23e101252f13